File: react_frontend_client/src/services/ProductService.js
import axios from 'axios';

const PRODUCT_API_BASE_URL = "http://localhost:8080/api/product"

class ProductService {

    getMainPageProducts() {
        return axios.get(PRODUCT_API_BASE_URL + '/main-page');
    }

    getRecommendedProducts() {
        return axios.get(PRODUCT_API_BASE_URL + '/recommend');
    }

    getTrendingProducts() {
        return axios.get(PRODUCT_API_BASE_URL + '/trending');
    }

    getProductById(id) {
        return axios.get(PRODUCT_API_BASE_URL + '/' + id);
    }

}

export default new ProductService()

File: server/src/main/java/com/usedgoods/server/service/ProductService.java
package com.usedgoods.server.service;

import com.usedgoods.server.model.Product;

import java.util.List;

public interface ProductService {
    Product saveProduct(Product product);
    Product getProductById(Long id);

    List<Product> getProducts();

    List<Product> getRecommendedProducts();

    List<Product> getMainPageProducts();

    List<Product> getTrendingProducts();
}
